akka-http-backend/src/test/scala/com/softwaremill/sttp/akkahttp/AkkaHttpRouteBackendTest.scala

```diff
@@ -0,0 +1,107 @@
+package com.softwaremill.sttp.akkahttp
+
+import akka.http.scaladsl.server.Route
+import akka.http.scaladsl.testkit.ScalatestRouteTest
+import akka.actor.ActorSystem
+import akka.http.scaladsl.model.{HttpRequest, HttpResponse}
+import akka.http.scaladsl.settings.ConnectionPoolSettings
+import com.softwaremill.sttp.SttpBackend
+import org.scalatest.{AsyncWordSpec, Matchers}
+
+import scala.concurrent.{ExecutionContext, Future, Promise}
+import scala.concurrent.duration._
+import akka.http.scaladsl.testkit.RouteTestTimeout
+
+class AkkaHttpRouteBackendTest extends AsyncWordSpec with ScalatestRouteTest with Matchers {
+
+  implicit val timeout = RouteTestTimeout(5.seconds)
+
+  lazy val backend: SttpBackend[Future, Nothing] = {
+    AkkaHttpBackend.usingActorSystem(system) {
+      new AkkaHttpClient {
+        override def singleRequest(request: HttpRequest, settings: ConnectionPoolSettings): Future[HttpResponse] = {
+          val p = Promise[HttpResponse]()
+          val f = p.future
+
+          request ~> Route.seal(Routes.route) ~> check {
```

Comment:
That's actually not how you would create a handler function from a route. This API is only used as a testing API.

The straight-forward way is using `Route.asyncHandler`:

https://github.com/akka/akka-http/blob/f57ec755a9107a8c1c25412ecde3149a91261e7c/akka-http/src/main/scala/akka/http/scaladsl/server/Route.scala#L71-L78

Reply:
This is great! Thanks.